junit-plugin/src/main/scala/org/scalajs/junit/plugin/ScalaJSJUnitPlugin.scala

```diff
@@ -0,0 +1,522 @@
+package org.scalajs.junit.plugin
+
+import scala.language.reflectiveCalls
+
+import scala.reflect.internal.Flags
+import scala.tools.nsc._
+import scala.tools.nsc.plugins.{
+  Plugin => NscPlugin, PluginComponent => NscPluginComponent
+}
+
+/** The Scala.js jUnit plugin is a way to overcome make class annotation
+ * information (usually accessed by reflection) of any test class accessible
+ * to the Scala.js testing framework.
+ *
+ *
+ * As an example we take the following test class:
+ * {{{
+ * class Foo {
+ *   @Before def before(): Unit = {
+ *     // Initialize the instance before the tests
+ *   }
+ *   @Test def bar(): Unit = {
+ *     // assert some stuff
+ *   }
+ *   @Ignore("baz not implemented yet") @Test def baz(): Unit = {
+ *     // assert some other stuff
+ *   }
+ * }
+ * object Foo {
+ *   @BeforeClass def beforeClass(): Unit = {
+ *     // Initialize some global state for the tests.
+ *   }
+ * }
+ * }}}
+ *
+ * Will be transformed to:
+ *
+ * {{{
+ * class Foo extends org.scalajs.junit.ScalaJSJUnitTest {
+ *   @Before def before(): Unit = {
+ *     // Initialize the instance before the tests
+ *   }
+ *   @Test def bar(): Unit = {
+ *     // assert some stuff
+ *   }
+ *   @Ignore @Test def baz(): Unit = {
+ *     // assert some other stuff
+ *   }
+ *   def scalajs\$junit\$invoke(methodId: String): Unit = {
```

Comment:
Maybe it would make sense to add this method to the `bootstrapper` instead, as:
```scala
def invoke(instance: AnyRef, methodId: String): Unit = {
  if (methodId == "0") instance.asInstanceOf[Foo].before()
  ...
}
```
That would allow to get rid of `extends org.scalajs.junit.ScalaJSJUnitTest`, which would probably simplify the transformation/make it more robust. Actually, I think in that case we don't need to modify `Foo` nor its companion object altogether. That would be nice, I think.